src/main/java/com/iota/iri/cache/impl/CacheImpl.java

```diff
@@ -100,6 +100,9 @@ public int getSize() {
 
     @Override
     public void put(K key, V value) {
+        if (key == null || value == null) {
+            return;
```

Comment:
Something is strange to me... if we look at `get()` we see
``` java
if (value == null && weakStore.containsKey(key)) {
            value = weakStore.get(key);
            put(key, value);
            weakStore.remove(key);
        }
```
This `put` failed meaning that that the condition returned true...
I guess what happened is that `weakStore.containsKey(key)` returned true -> GC kicked in -> failure at put

I am afraid that you are just masking the problem here... paving the way to more latent bugs
I would say that fixing `get()` as so would be better:

``` java
if (value == null) {
            value = weakStore.get(key);
            if (value != null) {
               put(key, value);
               weakStore.remove(key);
        }
```

Reply:
perhaps it is worth checking if at other places we forget that GC can kick in...

Reply:
Makes sense.

Reply:
> I am afraid that you are just masking the problem here
I would delete this change

Even though you are preventing, the error from happening you are just hiding the underlying cause
I would rather have an exception thrown in testing so we can do a proper fix when this happens